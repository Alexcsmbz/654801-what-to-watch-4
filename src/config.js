export const moviePageTabs = [
  {
    name: `Overview`,
  },
  {
    name: `Details`,
  },
  {
    name: `Reviews`,
  },
];

export const RoutePath = {
  MAIN: `/`,
  LOGIN: `/login`,
  MOVIE: `/films/:id`,
  MYLIST: `/mylist`,
  REVIEW: `/films/:id/review`,
};
