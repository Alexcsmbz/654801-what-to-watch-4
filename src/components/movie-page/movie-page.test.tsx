import {BrowserRouter as Router} from 'react-router-dom';
import MoviePage from './movie-page';
import {testMovie, testUser} from 'config';

describe(`MoviePage snapshot test`, () => {
  it(`MoviePage should render movie page`, () => {
    const tree = renderer.create(
        <Router>
          <MoviePage
            movies={[]}
            onMovieCardClick={() => {}}
            activeIdx={0}
            setActiveIdx={() => {}}
            isFullscreen={false}
            setIsFullscreen={() => {}}
            movie={testMovie}
            openFullscreen={() => {}}
            isAuth={true}
            user={testUser}
            toggleMovieInList={() => {}}
            addedMovies={[]}
            getCommentList={() => {}}
            commentList={[]}
          />
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
