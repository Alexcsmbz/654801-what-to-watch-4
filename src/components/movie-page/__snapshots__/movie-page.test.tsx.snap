// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage snapshot test MoviePage should render movie page 1`] = `
Array [
  <div
    className="visually-hidden"
  >
    <video
      controls={true}
      height="175"
      width="280"
    >
      <source
        src="videoLink"
        type="video/mp4"
      />
    </video>
  </div>,
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
        style={
          Object {
            "backgroundColor": "backgroundColor",
          }
        }
      >
        <img
          alt="name"
          src="backgroundImage"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link undefined"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <div
            className="user-block__avatar"
          >
            <a
              href="/mylist"
              onClick={[Function]}
            >
              <img
                alt="name avatar"
                height="63"
                src="https://4.react.pages.academyavatarUrl"
                width="63"
              />
            </a>
          </div>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            name
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              genre
            </span>
            <span
              className="movie-card__year"
            >
              2002
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn--play btn movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn--list btn movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <button
              className="btn movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <span>
                Add review
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="name"
            height="327"
            src="posterImage"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active "
                onClick={[Function]}
              >
                <div
                  className="movie-nav__link"
                >
                  Overview
                </div>
              </li>
              <li
                className="movie-nav__item undefined"
                onClick={[Function]}
              >
                <div
                  className="movie-nav__link"
                >
                  Details
                </div>
              </li>
              <li
                className="movie-nav__item undefined"
                onClick={[Function]}
              >
                <div
                  className="movie-nav__link"
                >
                  Reviews
                </div>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              8.8
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Very good
              </span>
              <span
                className="movie-rating__count"
              >
                370881
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              description
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                director
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                and other...
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      />
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
